#164798004 postgre sql setup

amaechi-chuks opened the pull request:
#### What does this PR do?
This PR setup PostgreSQL database

#### Description of Task to be completed?
- install express
- install sequelize
- install Postgres

#### How should this be manually tested?
Clone this repo https://github.com/andela/hermes-ah-backend/ into any folder of your choice
Cd into that folder
Checkout to ch-PostgreSQL-setup-#164798004 branch
Run `npm run start:dev` to run the server
#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164798004

rotimi-babalola commented on .env-example:
```diff
@@ -0,0 +1,5 @@
+DB_PASSWORD,
+DEV_DATABASE,
+DB_HOST,
+DB_PORT,
```
Add new line. Also you can remove the commas in the variable names

rotimi-babalola commented on .sequelizerc:
```diff
@@ -0,0 +1,8 @@
+  "models-path": path.resolve('./server/database/models'),
+  "seeders-path": path.resolve('./server/database/seeders'),
+  "migrations-path": path.resolve('./server/database/migrations')
+};
```
Add new line

rotimi-babalola commented on .gitignore:
```diff
@@ -0,0 +1,66 @@
+# build
+server/build
+.DS_Store
+dist/
```
Why am I seeing this file again? I thought we've already merged this 🤔

rotimi-babalola commented on server/app.js:
```diff
@@ -0,0 +1,21 @@
+app.use(cors());
+app.use(express.json());
+// support parsing of application/x-www-form-urlencoded post data
+app.use(bodyParser.urlencoded({ extended: true }));
```
You don't need to install `bodyParser` for this. You can use `express`. Look at using [express.urlencoded](https://expressjs.com/en/api.html#express.urlencoded)

rotimi-babalola commented on server/app.js:
```diff
@@ -0,0 +1,21 @@
+  // eslint-disable-next-line no-console
+  console.log(`App is listen on ${port}`);
+  db.sequelize.sync();
+});
```
Add new line and what is the use of `db.sequelize.sync`?

rotimi-babalola commented on server/app.js:
```diff
@@ -0,0 +1,17 @@
+app.listen(port, () => {
+  // eslint-disable-next-line no-console
+  console.log(`App is listen on ${port}`);
+  db.sequelize.sync();
```
Why are you doing this? - `db.sequelize.sync()` and also you left out using `express.urlencoded`

amaechi-chuks replied:
From the material you sent to me yesterday, I found out that body-parser is not needed because I was not making any request at the moment.

The db.sequelize.sync() function is used to show that migration has been established in the database.

rotimi-babalola replied:
I don't think you need to use `db.sequelize.sync` if you want to run migrations when starting up  the server you can create an npm script to do that and add it the `npm start` command for starting the server @amaechi-chuks

amaechi-chuks replied:
Noted.

rotimi-babalola commented:
You also need to fix resolve HoundCI errors before this PR can be merged @amaechi-chuks

rotimi-babalola commented:
Also I'm curious why you have `eslintrc.json` and `.gitignore` as changes in this PR. We already have these files in `develop` why are they appearing here again?

amaechi-chuks commented:
> Also I'm curious why you have `eslintrc.json` and `.gitignore` as changes in this PR. We already have these files in `develop` why are they appearing here again

amaechi-chuks commented:
> > Also I'm curious why you have `eslintrc.json` and `.gitignore` as changes in this PR. We already have these files in `develop` why are they appearing here again

amaechi-chuks commented:
> Also I'm curious why you have `eslintrc.json` and `.gitignore` as changes in this PR. We already have these files in `develop` why are they appearing here again?

When I was setting up PostgreSQL, I added a .gitignore file in my branch ch-PostgreSQL-setup-#164798004. The changes occur during the rebasing with setup eslint branch. 

In regards to eslintrc.json changes, I was having a blocker with my .babelrc, I was getting a 'module import is undefined', I initially thought it might be an error with our eslint, so I added that rule. However that didn't work so I consulted my teammates they all looked at it and eventually, we realized the babel file was placed on the wrong folder. I forgot to then remove that rule. Sorry about that.

rotimi-babalola commented on .eslintrc.json:
```diff
@@ -0,0 +1,40 @@
+      }]
+    }
+}
+
```
Again why is this file showing as a change? We already have this file in `develop` so why is showing up as a change?
